### PR TITLE
i#3049 bug-drwrap-exit: drwrap not clean up when exit

### PR DESCRIPTION
**Pull request by Louis-Ye**
1) Add code to clean all resources allocated on drwrap_init

Fixes [i3049](https://github.com/DynamoRIO/dynamorio/issues/3049)

### CODE REVIEW

File: ext/drwrap/drwrap.c [hunk 1]
```diff
@@ -974,6 +974,13 @@ drwrap_exit(void)
     if (count != 0)
         return;
 
+    drmgr_unregister_bb_app2app_event(drwrap_event_bb_app2app);
```
Comment (derekbruening) [inline ext/drwrap/drwrap.c]: Check the return value of the drmgr ones.

Comment (derekbruening) [inline ext/drwrap/drwrap.c]: I don't understand how these affect stability: while it is good to clean up, I thought they did not matter to the bottom line when exiting as drmgr and DR throw out the registration lists at exit time anyway.  Please update #3049 with details on why failing to unregister at exit time causes a problem.

File: ext/drwrap/drwrap.c [hunk 2]
```diff
@@ -974,6 +974,13 @@ drwrap_exit(void)
     if (count != 0)
```
Comment (derekbruening) [inline ext/drwrap/drwrap.c]: - In the title: s/i3/i#3/.
- In the title: if it's a single-commit issue, there is no mini-summary before the colon.
- In the description: s/Fixes i/Fixes #/; s/1.//.
- Please augment the description with what kind of resources (unregistering for events) and why it matters.

Comment (Louis-Ye) [reply ext/drwrap/drwrap.c]: Got it!

File: ext/drwrap/drwrap.c (hunk 1 above)
Comment (Louis-Ye) [reply ext/drwrap/drwrap.c]: I have updated #3049 with more details.

File: ext/drwrap/drwrap.c [hunk 3]
```diff
@@ -974,6 +974,17 @@ drwrap_exit(void)
...
         return;
 
+    ASSERT(drmgr_unregister_bb_app2app_event(drwrap_event_bb_app2app),
```
Comment (derekbruening) [inline ext/drwrap/drwrap.c]: Do the call outside the ASSERT: as is, this call will disappear in release build.  Ditto below.

**Pull request merged**